### PR TITLE
Configure logging

### PR DESCRIPTION
**Pull request by adfost**
Add Tenant Logs view and edit modal
Screen Shot 2021-10-28 at 1 25 56 PM

### CODE REVIEW

**Comment by jinapurapu** [comment]
GET is being called twice when loading Logs tab
Screenshot from 2021-11-04 14-51-59

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: portal-ui/src/screens/Console/Tenants/TenantDetails/TenantDetails.tsx [hunk 1]
```diff
@@ -222,6 +223,7 @@ const TenantDetails = ({
       case "metrics":
       case "license":
       case "security":
+      case "tenantLogs":
```
[the remark this answers is not in the document]
Comment (adfost) [reply portal-ui/src/screens/Console/Tenants/TenantDetails/TenantDetails.tsx]: fixed

**Review by dvaldivia — DISMISSED** [review]

**Review by bexsoft — CHANGES_REQUESTED** [review]
Please review all warnings in yarn server.

File: portal-ui/src/screens/Console/Tenants/TenantDetails/EditTenantLogsModal.tsx [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+    return retval;
+  };
+
+  const cleanValidation = (fieldName: string) => {
+    setValidationErrors(clearValidationError(validationErrors, fieldName));
+  };
+
+  useEffect(() => {
+    let tenantLogValidation: IValidation[] = [];
+    tenantLogValidation = [...tenantLogValidation];
```
Comment (dvaldivia) [inline portal-ui/src/screens/Console/Tenants/TenantDetails/EditTenantLogsModal.tsx:150]: This doesn't make any sense, remove it

File: portal-ui/src/screens/Console/Tenants/TenantDetails/KeyPairEdit.tsx [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        required: false,
+        value: newValues[i].value,
+        pattern: valuePattern,
+        customPatternMessage: "Invalid value",
+      });
+    }
+
+    const commonVal = commonFormValidation(keyPairValidation);
+    setError(commonVal);
+    console.log(error);
```
Comment (dvaldivia) [inline portal-ui/src/screens/Console/Tenants/TenantDetails/KeyPairEdit.tsx:109]: Remove this console log

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: portal-ui/src/screens/Console/Tenants/TenantDetails/KeyPairEdit.tsx (hunk 3 above)
Comment (adfost) [reply portal-ui/src/screens/Console/Tenants/TenantDetails/KeyPairEdit.tsx]: done

File: portal-ui/src/screens/Console/Tenants/TenantDetails/EditTenantLogsModal.tsx (hunk 2 above)
Comment (adfost) [reply portal-ui/src/screens/Console/Tenants/TenantDetails/EditTenantLogsModal.tsx]: done

**Review by bexsoft — APPROVED** [review]

**Comment by dvaldivia** [comment]
it doesn't let me create a three character service account called `uno` which kubernetes does let me create, thus it's a valid `sa` name

Screen Shot 2021-11-30 at 11 14 17 AM
Screen Shot 2021-11-30 at 11 13 26 AM

File: portal-ui/src/screens/Console/Tenants/TenantDetails/TenantLogs.tsx [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+                          </td>
+                          <td>{logInfo?.diskCapacityGB}</td>
+                        </tr>
+                        <tr>
+                          <td className={classes.titleCol}>
+                            Service Account Name:
+                          </td>
+                          <td>{logInfo?.serviceAccountName}</td>
+                        </tr>
+                        <h4>Labels</h4>
```
Comment (dvaldivia) [inline portal-ui/src/screens/Console/Tenants/TenantDetails/TenantLogs.tsx:282]: `Warning: validateDOMNesting(...): <h4> cannot appear as a child of <tbody>.`

Comment (dvaldivia) [reply portal-ui/src/screens/Console/Tenants/TenantDetails/TenantLogs.tsx:282]: all these html elements must be inside a column, it's invalid html if they are inside the body of the table

**Review by dvaldivia — CHANGES_REQUESTED** [review]

**Comment by dvaldivia** [comment]
Please fix all this browser warnings
Screen Shot 2021-11-30 at 11 19 40 AM
Screen Shot 2021-11-30 at 11 19 49 AM
Screen Shot 2021-11-30 at 11 19 46 AM

**Review by bexsoft — CHANGES_REQUESTED** [review]
It would be nice if delete button gets disabled when only one option is set
Screen Shot 2021-11-30 at 13 28 43

Also found another issue, when you type fast in these fields, DOM gets frozen for a couple of seconds and then all the text appears suddenly


https://user-images.githubusercontent.com/33497058/144115258-3c99bbb8-3a1d-4bb1-97a3-d440c5a6b2e6.mp4


:

**Comment by Alevsk** [comment]
@adfost please check why the request is triggering twice when someone click on the logs menu

Screen Shot 2021-11-30 at 14 05 33

I have seen a similar behavior on other sections of the application, we are doing exactly the same request twice

File: operatorapi/operator_tenants.go [hunk 5]
```diff
@@ -1706,6 +1740,299 @@ func getTenantUsageResponse(session *models.Principal, params operator_api.GetTe
...
+	}
+	dbNodeSelector := []*models.NodeSelector{}
+	for k, v := range minTenant.Spec.Log.Db.NodeSelector {
+		dbNodeSelector = append(dbNodeSelector, &models.NodeSelector{Key: k, Value: v})
+	}
+
+	if minTenant.Spec.Log.Audit == nil || minTenant.Spec.Log.Audit.DiskCapacityGB == nil {
+		minTenant.Spec.Log.Audit = &miniov2.AuditConfig{DiskCapacityGB: swag.Int(0)}
+	}
+
```
Comment (Alevsk) [inline operatorapi/operator_tenants.go:1801]:
```suggestion
	if minTenant.Spec.Log.Image == "" {
		minTenant.Spec.Log.Image = miniov2.DefaultLogSearchAPIImage
	}

	if minTenant.Spec.Log.Db.Image == "" {
		minTenant.Spec.Log.Db.Image = miniov2.LogPgImage
	}
```

Make sure to display the default logSearchApi and Postgres image if the user didn't defined custom images

**Review by Alevsk — CHANGES_REQUESTED** [review]

**Comment by Alevsk** [comment]
When logsearchApi and Postgres are enabled but user didn't defined an image explicitly we are displaying a blank value in the UI

Screen Shot 2021-11-29 at 20 38 36
Screen Shot 2021-11-30 at 14 24 51

We should display display the default

File: operatorapi/operator_tenants.go (hunk 5 above)
Comment (Alevsk) [reply operatorapi/operator_tenants.go:1804]: With this change now the image version is displayed:

Screen Shot 2021-11-30 at 14 40 21
Screen Shot 2021-11-30 at 14 40 28
Screen Shot 2021-11-30 at 14 40 31

Besides that the feature works fine.

**Comment by dvaldivia** [comment]
my tenant doesn't specify a log search image, if I edit and save, the default image is stored in the tenant CR, it shouldn't so that this version is updated with the operator

```
  log:
    audit:
      diskCapacityGB: 5
    db:
      resources: {}
      volumeClaimTemplate:
        metadata:
          creationTimestamp: null
          name: videos-log
        spec:
          accessModes:
          - ReadWriteOnce
          resources:
            requests:
              storage: "5368709120"
          storageClassName: standard
        status: {}
    resources: {}
```

**Review by dvaldivia — APPROVED** [review]

**Review by bexsoft — APPROVED** [review]

**Pull request merged**